From 39191374f832a5b44bd44a3072145d0c76d060da Mon Sep 17 00:00:00 2001
From: Michael Liao <michael.liao@intel.com>
Date: Fri, 19 Oct 2012 23:40:49 -0700
Subject: [PORT FROM MAIN] Export expf() and powf() standard C math functions

BZ: 66242

Cherry-pick from https://otc-android.intel.com/gerrit/#/c/16034

Change-Id: I81f63fca4a1921b1d17aaffb6aaf362dee690e38
Orig-Change-Id: Icf4a272f0544e780037cbc7e203c6f682ad77e13
Author: Michael Liao <michael.liao@intel.com>
Signed-off-by: Yong Chen <yong.a.chen@intel.com>
Reviewed-on: http://android.intel.com:8080/72627
Reviewed-by: lab_cactus <lab_cactus@intel.com>
Reviewed-by: Ren, Jack <jack.ren@intel.com>
Tested-by: Ren, Jack <jack.ren@intel.com>
Reviewed-by: Li, XiaojingX <xiaojingx.li@intel.com>
Tested-by: Li, XiaojingX <xiaojingx.li@intel.com>
Reviewed-by: buildbot <buildbot@intel.com>
Tested-by: buildbot <buildbot@intel.com>
Reviewed-on: http://android.intel.com:8080/79496
Reviewed-by: Yao, Yong <yong.yao@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
---
 cpu_ref/rsCpuRuntimeMath.cpp |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

diff --git a/cpu_ref/rsCpuRuntimeMath.cpp b/cpu_ref/rsCpuRuntimeMath.cpp
index 04a80e0..aa4ace4 100644
--- a/cpu_ref/rsCpuRuntimeMath.cpp
+++ b/cpu_ref/rsCpuRuntimeMath.cpp
@@ -252,6 +252,11 @@ static RsdCpuReference::CpuSymbol gSyms[] = {
     { "_Z6rsRandff", (void *)&SC_randf2, true },
     { "_Z6rsFracf", (void *)&SC_frac, true },
 
+    // Standard C library interfaces
+    { "expf", (void *)&expf, true },
+    { "powf", (void *)&powf, true },
+    { "exp2f", (void *)&exp2f, true },
+
     { NULL, NULL, false }
 };
 
-- 
1.7.4.1

